src/fundus/parser/utility.py

```diff
@@ -121,14 +121,23 @@ def extract_nodes(selector: XPath, node_type: Type[Node]) -> List[Node]:
 _meta_node_selector = CSSSelector("meta[name], meta[property]")
 
 
-def get_meta_content(tree: lxml.html.HtmlElement) -> Dict[str, str]:
+def get_meta_content(
+    tree: lxml.html.HtmlElement, additional_selectors: Optional[Dict[str, Union[XPath, CSSSelector]]] = None
+) -> Dict[str, str]:
     meta_nodes = _meta_node_selector(tree)
     meta: Dict[str, str] = {}
     for node in meta_nodes:
         key = node.attrib.get("name") or node.attrib.get("property")
         value = node.attrib.get("content")
         if key and value:
             meta[key] = value
+
+    if additional_selectors:
+        for key, selector in additional_selectors.items():
+            for node in selector(tree):
+                if content := node.attrib.get("content"):
```

Comment:
There is an edge case, if there are multiple meta tags using the attribute within the selector. For example using: a document with html: `'<meta itemprop="headline" content="Erster Inhalt"><meta itemprop="headline" content="Zweiter Inhalt"><meta name="robots" content="max-image-preview:large"/>'` and calling `get_meta_content(lxml.html.document_fromstring(html), dict({"itemprop": lxml.cssselect.CSSSelector("meta[itemprop]")}))` will get you the result: `{'headline': 'Zweiter Inhalt', 'robots': 'max-image-preview:large'}`. I'm not sure how meta tags are restricted, but in case there's the possibility of them having the same attribute value, this might lead to unwanted behavior.

Reply:
Within the meta `name`/`property`/`itemprop` building three different categories. `name` coming from HTML5, `property` from RDFa, and god knows where `itemprop` is coming from. Within those categories having duplicate names shouldn't be allowed and overriding seems to be an appropriate strategy here.